apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,165 @@
+import {useSettings} from '@yearn-finance/web-lib/contexts/useSettings';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {useChainID} from '@yearn-finance/web-lib/hooks/useChainID';
+import {toAddress} from '@yearn-finance/web-lib/utils/address';
+import {ETH_TOKEN_ADDRESS} from '@yearn-finance/web-lib/utils/constants';
+import {defaultTxStatus} from '@yearn-finance/web-lib/utils/web3/transaction';
+import {useWallet} from '@common/contexts/useWallet';
+import {useAllowanceFetcher} from '@common/hooks/useAllowanceFetcher';
+import {DefaultTNormalizedBN} from '@common/utils';
+
+import type {ReactElement} from 'react';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+import type {TAllowanceFetcher} from '@common/hooks/useAllowanceFetcher';
+import type {TNormalizedBN} from '@common/types/types';
+
+function	VaultDetailsQuickActionsButtons(): ReactElement {
+	const {refresh} = useWallet();
+	const {isActive} = useWeb3();
+	const {safeChainID} = useChainID();
+	const {networks} = useSettings();
+	const [txStatusApprove, set_txStatusApprove] = useState(defaultTxStatus);
+	const [txStatusDeposit, set_txStatusDeposit] = useState(defaultTxStatus);
+	const [allowanceFrom, set_allowanceFrom] = useState<TNormalizedBN>();
+	const {
+		selectedOptionFrom, selectedOptionTo,
+		amount, onChangeAmount,
+		maxDepositPossible, isDepositing
+	} = useActionFlow();
+	const {onApprove, onExecuteDeposit, onExecuteWithdraw, currentSolver} = useSolver();
+	const retrieveAllowance = useAllowanceFetcher();
+
+	const shouldUseVanillaAllowance = [Solvers.VANILLA, Solvers.CHAIN_COIN, Solvers.PARTNER_CONTRACT].includes(currentSolver);
+	const isInputTokenEth = selectedOptionFrom?.value === ETH_TOKEN_ADDRESS;
+	const isOutputTokenEth = selectedOptionTo?.value === ETH_TOKEN_ADDRESS;
+	const canInteract = isActive && amount.raw.gt(0) && selectedOptionFrom && selectedOptionTo;
+
+	const spender = useMemo((): TAddress => {
+		if (currentSolver === Solvers.CHAIN_COIN && isOutputTokenEth) {
+			return (toAddress(getEthZapperContract(safeChainID)));
+		} else if (currentSolver === Solvers.PARTNER_CONTRACT) {
+			return (toAddress(networks?.[safeChainID]?.partnerContractAddress));
+		}
+		return (toAddress(selectedOptionTo?.value));
+	}, [currentSolver, isOutputTokenEth, networks, safeChainID, selectedOptionTo?.value]);
+
+	/* 🔵 - Yearn Finance **************************************************************************
+	** SWR hook to get the expected out for a given in/out pair with a specific amount. This hook is
+	** called every 10s or when amount/in or out changes. Calls the allowanceFetcher callback.
+	**********************************************************************************************/
+	const	{data: vanillAllowanceFrom, isLoading: isValidatingAllowance, mutate: mutateAllowance} = useSWR(
+		canInteract && !isInputTokenEth && shouldUseVanillaAllowance ?
+		[selectedOptionFrom, spender] as TAllowanceFetcher : null,
```

Comment:
Is this casting required?
```suggestion
		[selectedOptionFrom, spender] : null,
```